rfcs/0190-queue-change-task-run-priority.md

```diff
@@ -0,0 +1,55 @@
+# RFC 0190 - Queue change task run priority
+* Comments: [#0190](https://github.com/taskcluster/taskcluster-rfcs/pull/190)
+* Proposed by: @lotas
+
+## Summary
+
+Since we refactored queue internals it became possible to see what tasks are pending for a particular
+worker pool / task queue.
+
+This RFC proposes new API method that would allow to change the priority of the existing task run.
+
+## Motivation
+
+There might be several use-cases when some worker pool has a lot of pending tasks and there is a need
+to move some more important task to the top of the waiting list, or also move down less important ones.
+
+Instead of cancelling existing tasks we can instead choose to change task run priorities.
+
+## Details
+
+I propose we implement this on a task run level and not the task itself:
+
+* tasks are supposed to be immutable
+* changing the priority makes more sense in the context of a specific run, not the whole task
+
+Queue service will expose new method `queue.changeTaskRunPriority(taskId, runId, newPriority)`.
```

Comment:
Any thoughts of a batch API as well? I could envision the pending task list be bulk editable, where you select multiple tasks that you want to increase/decrease the priority of. @Archaeopteryx @jcristau would this be something you guys would use or am I thinking of a non-issue here?

Reply:
I'd like to +1 this. As mentioned in the community meeting there are definitely use cases where an entire task group should be bumped in priority.